torch/csrc/jit/script/init.cpp

```diff
@@ -527,7 +534,13 @@ void initJitScriptBindings(PyObject* module) {
   });
 
   m.def("merge_type_from_type_comment", &mergeTypesFromTypeComment);
-  m.def("import_ir_module", import_ir_module);
+  m.def("import_ir_module", [](ModuleLookup module_lookup, const std::string& filename) {
+      import_ir_module(module_lookup, filename);
+      });
+  m.def("import_ir_module_from_buffer", [](ModuleLookup module_lookup, const std::string& buffer) {
+      std::istringstream in(buffer);
```

Comment:
This is unfortunately extremely inefficient, because it will keep 3 copies of the buffer in memory: one for the Python bytes object, one for `std::string` (which needs to own its data), and one for `istringstream` which uses `buffer` to fill its internal buffer. I'm not sure how this can be fixed though.

Reply:
I've tried looking for a solution as well, but I'm short of ideas if we still want to go with streams.